apps/src/templates/studioHomepages/UiTips.jsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import UiTip from './UiTip';
+import Dialog from '../Dialog';
+
+const UiTips = React.createClass({
+  propTypes: {
+    userId: React.PropTypes.number,
+    tipId: React.PropTypes.string,
+    showInitialTips: React.PropTypes.bool,
+    beforeDialog: React.PropTypes.object,
```

Comment:
You can specify these more than just being an object. In fact, it's easy to just reference the propTypes of the Dialog component:

```jsx
  propTypes: {
    ...
    beforeDialog: React.PropTypes.shape(Dialog.propTypes),
    afterDialog: React.PropTypes.shape(Dialog.propTypes),
  }
```